### ISSUE TITLE
False positive on MaxChainedCallsOnSameLine

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#5598]
Detekt 1.22.0 using Jetbrains Compose 1.3.0-beta03 (the `actual` is because I want to use this from common code, but other targets besides jvm don't have an `onClick` modifier).

The error is:

```
8 chained calls on a single line; more than 5 calls should be wrapped to a new line. [MaxChainedCallsOnSameLine]
```
and it occurs at the first `nextModifier` call.

```
@OptIn(ExperimentalFoundationApi::class)
actual fun Modifier.onDesktopMouseEvent(
  onMouseLeftClick: (() -> Unit)?,
  onMouseLeftDoubleClick: (() -> Unit)?,
  onMouseLeftLongClick: (() -> Unit)?,
  onMouseRightClick: (() -> Unit)?,
  onMouseRightDoubleClick: (() -> Unit)?,
  onMouseRightLongClick: (() -> Unit)?,
  onMouseMiddleClick: (() -> Unit)?,
  onMouseMiddleDoubleClick: (() -> Unit)?,
  onMouseMiddleLongClick: (() -> Unit)?,
  onMouseHorizontalRightScroll: (() -> Unit)?,
  onMouseHorizontalRightFling: (() -> Unit)?,
  onMouseHorizontalLeftScroll: (() -> Unit)?,
  onMouseHorizontalLeftFling: (() -> Unit)?,
  onMouseBackClick: (() -> Unit)?,
  onMouseBackDoubleClick: (() -> Unit)?,
  onMouseBackLongClick: (() -> Unit)?,
  onMouseForwardClick: (() -> Unit)?,
  onMouseForwardDoubleClick: (() -> Unit)?,
  onMouseForwardLongClick: (() -> Unit)?,
  onMouseThumbClick: (() -> Unit)?,
  onMouseThumbDoubleClick: (() -> Unit)?,
  onMouseThumbLongClick: (() -> Unit)?,
): Modifier {
  fun Modifier.nextModifier(
    button: PointerButton,
    click: (() -> Unit)?,
    doubleClick: (() -> Unit)?,
    longClick: (() -> Unit)?
  ) = if(click != null || doubleClick != null || longClick != null) {
    onClick(
      matcher = PointerMatcher.mouse(button),
      onClick = click ?: {},
      onDoubleClick = doubleClick,
      onLongClick = longClick
    )
  }
  else {
    this
  }

  return nextModifier(
    button = PointerButton(0),
    click = onMouseLeftClick,
    doubleClick = onMouseLeftDoubleClick,
    longClick = onMouseLeftLongClick
  ).nextModifier(
    button = PointerButton(1),
    click = onMouseRightClick,
    doubleClick = onMouseRightDoubleClick,
    longClick = onMouseRightLongClick
  ).nextModifier(
    button = PointerButton(2),
    click = onMouseMiddleClick,
    doubleClick = onMouseMiddleDoubleClick,
    longClick = onMouseMiddleLongClick
  ).nextModifier(
    button = PointerButton(3),
    click = onMouseHorizontalRightScroll,
    doubleClick = onMouseHorizontalRightFling,
    longClick = null
  ).nextModifier(
    button = PointerButton(4),
    click = onMouseHorizontalLeftScroll,
    doubleClick = onMouseHorizontalLeftFling,
    longClick = null
  ).nextModifier(
    button = PointerButton(5),
    click = onMouseBackClick,
    doubleClick = onMouseBackDoubleClick,
    longClick = onMouseBackLongClick
  ).nextModifier(
    button = PointerButton(6),
    click = onMouseForwardClick,
    doubleClick = onMouseForwardDoubleClick,
    longClick = onMouseForwardLongClick
  ).nextModifier(
    button = PointerButton(7),
    click = onMouseThumbClick,
    doubleClick = onMouseThumbDoubleClick,
    longClick = onMouseThumbLongClick
  )
}
```

**Comment by BraisGabin** [comment]
I'm not sure if this is a false positive.

I think that instead of 

```kotlin
).nextModifier(
```

It should be

```kotlin
)
.nextModifier(
```

**Comment by eygraber** [comment]
I think it's a stylistic choice. But in any case this is definitely not on the same line, so it's either a bug (which is my interpretation of what I would want this rule to do), or the rule should be renamed.

**Comment by atulgpt** [comment]
Hi, @cortinico / @BraisGabin I also faced this issue. I created the problem at https://github.com/detekt/detekt/issues/5696 which seems similar to this one. Is the handling of `\n` is expected solution? If yes I can pick this one

**Comment by BraisGabin** [comment]
Related with #5591

This rule was written as a companion of `CascadingCallWrapping`. Basically to avoid things like this one. There is even a test in that Rule to check that:

https://github.com/detekt/detekt/blob/b8cc640041ef1add4fbf04ed5b00cfa65fde67fe/detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/CascadingCallWrappingSpec.kt#L108-L125

But there are more problems. For example code like this:

```kotlin
val a = 0.plus(
    0).plus(
    0).plus(
    0)
```

The current implementation is really straight forward, implementing this will make the things more clomplex.

In my opinion implementing this makes this rule lose a lot of sense. This pair of rules doesn't care about how long is each line. There are other rules for that. They work like this: `CascadingCallWrapping` align how to wrap chained calls and `MaxChainedCallsOnSameLine` ensure that you don't cheat by doing this:

```kotlin
val a = 0.plus(
    0
  ).plus(
    0
  ).plus(
    0
  ).plus(
    0
  )
```

Or at least that is how I see them.

Maybe @dzirbel has some ideas regarding this as the author of the rule.

Anyway: if we implement this we should add a configuration to opt-in/opt-out.

**Comment by dzirbel** [comment]
I agree this is a bug in the rule, and it should ignore cases wrapped in this style where they're not actually on a single line. My intention with this rule was to break up chained calls where a ton of functionality is compressed into a single line, which is certainly not a problem in the initial example here or in general with chaining done via `\n).foo` rather than `)\n.foo`. It would make sense to me to have `CascadingCallWrapping` warn for that style (which is fine - it's a particular style choice which can be opted out of), but I think we can make `MaxChainedCallsOnSameLine` agnostic to the wrapping style.

**Comment by atulgpt** [comment]
Hi @dzirbel / @BraisGabin choosing `\n).foo` or `)\n.foo` should not be the concern of this rule `MaxChainedCallsOnSameLine`. If want to enforce consistency in a project either use `\n).foo` or `)\n.foo` we should create separate rule

**Comment by BraisGabin** [comment]
Ok, so this issue is ready to grab by anyone.

**Comment by atulgpt** [comment]
Hi, @BraisGabin I will work one it as my project is also impacted by this. Please assign this to me
